gobblin-metastore/src/main/java/gobblin/metastore/FsStateStore.java

```diff
@@ -281,7 +280,7 @@ public void createAlias(String storeName, String original, String alias) throws
     Path aliasTablePath = new Path(new Path(this.storeRootDir, storeName), alias);
     // Make a copy of the original table as a work-around because
     // Hadoop version 1.2.1 has no support for symlink yet.
-    FileUtil.copy(this.fs, originalTablePath, this.fs, aliasTablePath, false, true, this.conf);
+    HadoopUtils.copyPath(this.fs, originalTablePath, this.fs, aliasTablePath, this.conf);
```

Comment:
I agree that these methods should have an `overwrite` flag, but adding it is a more work than I though it would be. I think we would have to replace `FileUtil.copy`. The problem is that `FileUtil.copy` has a few pre-condition checks that it does and if any fail it throws an `IOException`. It also throws on `IOException` if there is a problem copying the actual data. The problem is that there is no way to differentiate the two scenarios.

One of the requirements of `HadoopUtils.copyPath` should be that it doesn't leave partial data - it doesn't copy part of a file, fail and then quit. This can cause `EOFException`s (which is the original motivation for this PR), which is why in the `catch` block the the `dst` is deleted. This is harder when `overwrite=true` as the method should not delete the `dst` if a pre-condition fails.